README.md

```diff
@@ -91,22 +91,26 @@ Share Social , Sending Simple Data to Other Apps
     and have it return the `android:authorities` that was added in
     AndroidManifest file.
   - For example: Replace the `com.example.yourappidhere` below with the
-    `applicationId` that is defined in your `android/app/build.gradle`. It must
-    be [hard-coded here to work
-    properly](https://github.com/EstebanFuentealba/react-native-share/issues/200#issuecomment-361938532).
+    `applicationId` that is defined in your `android/app/build.gradle`.
 
     ```
     import cl.json.ShareApplication
+    import android.content.Context;
 
     class MyApplication extends Application implements ShareApplication, ReactApplication {
+            private static Context context;
 
     {
 
          //...
+         @Override public void onCreate() {
+            //...
+            MyApplication.context = getApplicationContext();
+        }
 
          @Override
          public String getFileProviderAuthority() {
-                return "com.example.yourappidhere.provider";
+                return context.getPackageName() + ".provider";
```

Comment:
why not

`getApplicationContext().getPackageName() + ".provider"`?

Reply:
I have `this.getPackageName()` + ".provider" now